examples/protocol.py

```diff
@@ -0,0 +1,78 @@
+import argparse
+import asyncio
+import logging
+import os
+
+from prompt_toolkit import PromptSession
+
+import ferny
+
+
+class PromptResponder(PromptSession, ferny.AskpassHandler):
+    async def do_hostkey(self, reason: str, host: str, algorithm: str, key: str, fingerprint: str) -> bool:
+        print(f'Host key {host} {fingerprint}')
+        response = await self.prompt_async('Accept [yes/no/abort]? ')
+        if response.lower() in ['a', 'abort']:
+            raise RuntimeError('rejecting hostkey')
+        return response.lower() in ['y', 'yes']
+
+    async def do_askpass(self, messages: str, prompt: str, hint: str) -> str | None:
+        print(messages)
+        try:
+            print('--askpass--')
+            if hint == 'none':
+                print(prompt)
+                print('[ waiting ]')
+                while True:
+                    await asyncio.sleep(10)
+                return None
```

Comment:
This should get a comment. It looks like it would just sit here without *ever* actually returning from the do_askpass handler. If the function is merely to show the message (and no reply, as there is no hint), why is this necessary? This smells like a hack.

Reply:
This is a direct copy from the other example script, and it shows more of a limitation in Python's standard library than anything else: there's no simple way to wait forever.

The intent here is definitely that we do wait forever.  The way this works at the `ssh` level is the same: `ssh-askpass` is expected to present a prompt to the user and then never exit.  `ssh` will `kill()` `ssh-askpass` when it's done with it, and we'll turn that into a cancellation of the task that the handler function is running it.

It seemed like the most direct way to map the semantics, and I'm actually pretty happy with it.  We can use context managers or `finally:` blocks here to do cleanup as we deem appropriate.  It's a bit of a moot point at present, since we don't have a reasonable way to present such prompts in the Cockpit protocol, much less a way to retract them.

Reply:
I see. It just seems odd to wait forever and then having a `return` (which should be a "not reached"?) in a handler callback. That's what I mean with "comment", expecially as the example is something that most users will look at to learn how to use it.

Reply:
Fair enough.  I'll add a comment.